aesara/link/numba/dispatch/elemwise.py

```diff
@@ -280,9 +280,7 @@ def axis_apply_fn(x):
 
 @numba_funcify.register(CAReduce)
 def numba_funcify_CAReduce(op, node, **kwargs):
-    axes = op.axis
-    if axes is None:
-        axes = list(range(node.inputs[0].ndim))
+    axes = [int(_axis.data) for _axis in node.inputs[1:]]
```

Comment:
I believe JAX and Numba both operate under the assumption that the axis argument is always fixed (in their graphs), but the `Op` changes in this PR allow non-fixed/constant axis arguments in the equivalent Aesara `Op`s.

The implication is that the JAX/Numba transpilation code should check that the values are `Constant`s and not assume they are, because the Aesara `Op` doesn't.

Also, don't forget that `Apply` nodes can have their inputs changed in-place, which necessarily avoids the `Op.make_node` input checks.  This isn't anything to be too concerned about, unless someone creates a rewrite that intentionally replaces the axis arguments with non-`Constant` ones, but it's an important thing to consider regarding the design of this library and the expectations that other pieces of code should have.

Reply:
I added a check for that in the Numba transpilation code, however the jax one seems fine, because the axis argument is actually supported in jax hence we don't need to account for it being changed beforehand at type level.